homeassistant/components/ozw/lock.py

```diff
@@ -37,3 +75,48 @@ async def async_lock(self, **kwargs):
     async def async_unlock(self, **kwargs):
         """Unlock the lock."""
         self.values.primary.send_value(False)
+
+    def set_usercode(self, code_slot, usercode):
+        """Set the usercode to index X on the lock."""
+        lock_node = self.values._node.values()
+
+        for value in lock_node:
+            if (
+                value.command_class == CommandClass.USER_CODE
+                and value.index == code_slot
+            ):
+                if len(str(usercode)) < 4:
+                    _LOGGER.error(
+                        "Invalid code provided: (%s) user code must be at least 4 digits",
+                        usercode,
+                    )
+                    break
+                value.send_value(usercode)
+                break
+
+    def get_usercode(self, code_slot):
+        """Get a usercode at index X on the lock."""
+        lock_node = self.values._node.values()
+
+        for value in lock_node:
+            if (
+                value.command_class == CommandClass.USER_CODE
+                and value.index == code_slot
+            ):
+                _LOGGER.info("User code at slot %s is: %s", code_slot, value.value)
+                break
+
+    def clear_usercode(self, code_slot):
+        """Set usercode to slot X on the lock."""
+        lock_node = self.values._node.values()
+
+        for value in lock_node:
+            if (
+                value.command_class == CommandClass.USER_CODE
+                and value.label == "Remove User Code"
```

Comment:
Labels can change over time.

Reply:
The index is reported incorrectly by python-openzwave-mqtt as `ValueIndex.UNKNOWN` the actual ValueIndex should report `256`.